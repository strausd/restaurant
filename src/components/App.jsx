import React from 'react';
import { connect } from 'react-redux';

import AppRouter from '../routers/AppRouter';
import { fetchRestaurants } from '../actions/baseActions'


class App extends React.Component {
    constructor(props) {
        super(props);
        props.fetchRestaurants();
    }

    render() {
        return (
            <div>
                <AppRouter />
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchRestaurants: () => dispatch(fetchRestaurants())
    };
};

export default connect(undefined, mapDispatchToProps)(App);