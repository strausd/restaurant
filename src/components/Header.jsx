import React from 'react';
import { Link } from 'react-router-dom';


const Header = props => {
    const back_button = window.location.pathname !== '/';
    return (
        <div className="header">
            {back_button ? (
                <div className="header__icon header__back">
                    <Link to="/">
                        <img src="/dist/icons/ic_webBack@2x.png" alt="Map" />
                    </Link>
                </div>
            ) : null}
            <div className="header__link">
                <Link to="/">
                    <h1>Lunch Tyme</h1>
                </Link>
            </div>
            <div className="header__icon header__map">
                <Link to="/map">
                    <img src="/dist/icons/icon_map@2x.png" alt="Map" />
                </Link>
            </div>
        </div>
    );
};

export default Header;