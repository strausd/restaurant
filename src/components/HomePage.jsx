import React from 'react';
import { connect } from 'react-redux'

import RestaurantsList from './RestaurantsList';


export const HomePage = () => {
    return (
        <div className="page-content">
            <RestaurantsList />
        </div>
    );
};

export default HomePage;