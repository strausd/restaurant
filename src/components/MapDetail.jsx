import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps"

const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        <Marker position={{ lat: props.lat, lng: props.lng }}>
            <InfoWindow>
                <div>Testing</div>
            </InfoWindow>
        </Marker>
    </GoogleMap>
));

export default MapDetail;