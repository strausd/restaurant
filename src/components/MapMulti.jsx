import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps";

export const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        {props.restaurants.map((r, index) => {
            return (
                <Marker key={index} position={{ lat: r.lat, lng: r.lng }} onClick={() => props.onMarkerToggle(index)}>
                    {r.isOpenMarkerInfo && <InfoWindow onCloseClick={props.onMarkerToggle}>
                        <div className="marker-info">
                            <div className="marker-info__name">{r.name}</div>
                            <div className="marker-info__directions">
                                <a target="_blank" href={'https://www.google.com/maps/dir/?api=1&destination=' + r.gMapsQuery}>
                                    Get Directions
                                </a>
                            </div>
                        </div>
                    </InfoWindow>}
                </Marker>
            );
        })}
    </GoogleMap>
));

export default MapDetail;