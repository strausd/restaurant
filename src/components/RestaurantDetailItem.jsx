import React from 'react';

// AIzaSyCRGU4zvepa0suuU1BZu_tWiEn7ho5MSEc
const RestaurantDetailItem = ({ restaurant }) => {
    const destination_str = encodeURIComponent(
        restaurant.name + ' ' +
        restaurant.location.address + ' ' +
        restaurant.location.city + ' ' +
        restaurant.location.state + ' ' +
        restaurant.location.postalCode
    );
    return (
        <div className="restaurant-detail">
            <div className="restaurant-detail__map">

            </div>
            <div className="restaurant-detail__header">
                <h2>{restaurant.name}</h2>
                <div className="restaurant-detail__category">{restaurant.category}</div>
            </div>
            <div className="restaurant-detail__info">
                <div className="restaurant-detail__address">
                    <a target="_blank" href={'https://www.google.com/maps/dir/?api=1&destination=' + destination_str}>
                        {restaurant.location.address}
                        <br/>
                        {restaurant.location.city}, {restaurant.location.state} {restaurant.location.postalCode}
                    </a>
                </div>
                {
                    !restaurant.contact ? null : (
                        <div>
                            <div className="restaurant-detail__phone">
                                <a href={'tel:' + restaurant.contact.phone} >{restaurant.contact.formattedPhone}</a>
                            </div>
                            <div className="restaurant-detail__twitter">
                                <a href={'https://twitter.com/' + restaurant.contact.twitter} target="_blank" >@{restaurant.contact.twitter}</a>
                            </div>
                        </div>
                    )
                }
            </div>
        </div>
    );
};

export default RestaurantDetailItem;