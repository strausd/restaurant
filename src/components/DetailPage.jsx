import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import RestaurantDetailItem from './RestaurantDetailItem';


export class DetailPage extends React.Component {
    render() {
        const id = this.props.match.params.id;
        if (isNaN(id)) {
            return <Redirect to="/" />
        } else if (!this.props.restaurants || this.props.restaurants.length < 1) {
            return (
                <div className="page-content">
                    Loading...
                </div>
            );
        } else {
            const restaurant = this.props.restaurants[id];
            return (
                <div className="page-content">
                    <RestaurantDetailItem restaurant={restaurant} />
                </div>
            );
        }
    }
}


const mapStateToProps = ({ restaurants, restaurant }) => {
    return {
        restaurants,
        restaurant
    };
};

export default connect(mapStateToProps)(DetailPage);