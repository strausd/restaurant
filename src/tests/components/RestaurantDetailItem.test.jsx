import React from 'react';
import { shallow } from 'enzyme';

import { RestaurantDetailItem } from '../../components/RestaurantDetailItem';
import { restaurants } from '../testData/restaurants';


let wrapper, restaurant;

beforeAll(() => {
    restaurant = restaurants[0];
    wrapper = shallow(<RestaurantDetailItem restaurant={restaurant} />);
});

test('should render RestaurantDetailItem component properly', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().onMarkerToggle();
    expect(wrapper.state('isOpenMarkerInfo')).toBe(true);
});