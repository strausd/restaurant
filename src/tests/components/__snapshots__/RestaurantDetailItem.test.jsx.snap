// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RestaurantDetailItem component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantDetailItem
    restaurant={
      Object {
        "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
        "category": "Burgers",
        "contact": Object {
          "formattedPhone": "(972) 387-2337",
          "phone": "9723872337",
          "twitter": "hopdoddy",
        },
        "location": Object {
          "address": "5100 Belt Line Road, STE 502",
          "cc": "US",
          "city": "Addison",
          "country": "United States",
          "crossStreet": "Dallas North Tollway",
          "formattedAddress": Array [
            "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
            "Addison, TX 75254",
            "United States",
          ],
          "lat": 32.950787,
          "lng": -96.821118,
          "postalCode": "75254",
          "state": "TX",
        },
        "name": "Hopdoddy Burger Bar",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="restaurant-detail__map"
        >
          <withScriptjs(withGoogleMap(Component))
            containerElement={
              <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />
            }
            googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places"
            isMarkerShow={true}
            isOpenMarkerInfo={false}
            lat={32.950787}
            lng={-96.821118}
            loadingElement={
              <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />
            }
            mapElement={
              <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />
            }
            onMarkerToggle={[Function]}
            restaurant={
              Object {
                "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
                "category": "Burgers",
                "contact": Object {
                  "formattedPhone": "(972) 387-2337",
                  "phone": "9723872337",
                  "twitter": "hopdoddy",
                },
                "location": Object {
                  "address": "5100 Belt Line Road, STE 502",
                  "cc": "US",
                  "city": "Addison",
                  "country": "United States",
                  "crossStreet": "Dallas North Tollway",
                  "formattedAddress": Array [
                    "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
                    "Addison, TX 75254",
                    "United States",
                  ],
                  "lat": 32.950787,
                  "lng": -96.821118,
                  "postalCode": "75254",
                  "state": "TX",
                },
                "name": "Hopdoddy Burger Bar",
              }
            }
            zoom={13}
          />
        </div>,
        <div
          className="restaurant-detail__header"
        >
          <h2>
            Hopdoddy Burger Bar
          </h2>
          <div
            className="restaurant-detail__category"
          >
            Burgers
          </div>
        </div>,
        <div
          className="restaurant-detail__info"
        >
          <div
            className="restaurant-detail__address"
          >
            <a
              href="https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254"
              target="_blank"
            >
              5100 Belt Line Road, STE 502
              <br />
              Addison
              , 
              TX
               
              75254
            </a>
          </div>
          <div>
            <div
              className="restaurant-detail__phone"
            >
              <a
                href="tel:9723872337"
              >
                (972) 387-2337
              </a>
            </div>
            <div
              className="restaurant-detail__twitter"
            >
              <a
                href="https://twitter.com/hopdoddy"
                target="_blank"
              >
                @
                hopdoddy
              </a>
            </div>
          </div>
        </div>,
      ],
      "className": "restaurant-detail",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <withScriptjs(withGoogleMap(Component))
            containerElement={
              <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />
            }
            googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places"
            isMarkerShow={true}
            isOpenMarkerInfo={false}
            lat={32.950787}
            lng={-96.821118}
            loadingElement={
              <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />
            }
            mapElement={
              <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />
            }
            onMarkerToggle={[Function]}
            restaurant={
              Object {
                "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
                "category": "Burgers",
                "contact": Object {
                  "formattedPhone": "(972) 387-2337",
                  "phone": "9723872337",
                  "twitter": "hopdoddy",
                },
                "location": Object {
                  "address": "5100 Belt Line Road, STE 502",
                  "cc": "US",
                  "city": "Addison",
                  "country": "United States",
                  "crossStreet": "Dallas North Tollway",
                  "formattedAddress": Array [
                    "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
                    "Addison, TX 75254",
                    "United States",
                  ],
                  "lat": 32.950787,
                  "lng": -96.821118,
                  "postalCode": "75254",
                  "state": "TX",
                },
                "name": "Hopdoddy Burger Bar",
              }
            }
            zoom={13}
          />,
          "className": "restaurant-detail__map",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "containerElement": <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />,
            "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places",
            "isMarkerShow": true,
            "isOpenMarkerInfo": false,
            "lat": 32.950787,
            "lng": -96.821118,
            "loadingElement": <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />,
            "mapElement": <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />,
            "onMarkerToggle": [Function],
            "restaurant": Object {
              "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
              "category": "Burgers",
              "contact": Object {
                "formattedPhone": "(972) 387-2337",
                "phone": "9723872337",
                "twitter": "hopdoddy",
              },
              "location": Object {
                "address": "5100 Belt Line Road, STE 502",
                "cc": "US",
                "city": "Addison",
                "country": "United States",
                "crossStreet": "Dallas North Tollway",
                "formattedAddress": Array [
                  "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
                  "Addison, TX 75254",
                  "United States",
                ],
                "lat": 32.950787,
                "lng": -96.821118,
                "postalCode": "75254",
                "state": "TX",
              },
              "name": "Hopdoddy Burger Bar",
            },
            "zoom": 13,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Hopdoddy Burger Bar
            </h2>,
            <div
              className="restaurant-detail__category"
            >
              Burgers
            </div>,
          ],
          "className": "restaurant-detail__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hopdoddy Burger Bar",
            },
            "ref": null,
            "rendered": "Hopdoddy Burger Bar",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Burgers",
              "className": "restaurant-detail__category",
            },
            "ref": null,
            "rendered": "Burgers",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="restaurant-detail__address"
            >
              <a
                href="https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254"
                target="_blank"
              >
                5100 Belt Line Road, STE 502
                <br />
                Addison
                , 
                TX
                 
                75254
              </a>
            </div>,
            <div>
              <div
                className="restaurant-detail__phone"
              >
                <a
                  href="tel:9723872337"
                >
                  (972) 387-2337
                </a>
              </div>
              <div
                className="restaurant-detail__twitter"
              >
                <a
                  href="https://twitter.com/hopdoddy"
                  target="_blank"
                >
                  @
                  hopdoddy
                </a>
              </div>
            </div>,
          ],
          "className": "restaurant-detail__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254"
                target="_blank"
              >
                5100 Belt Line Road, STE 502
                <br />
                Addison
                , 
                TX
                 
                75254
              </a>,
              "className": "restaurant-detail__address",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "5100 Belt Line Road, STE 502",
                  <br />,
                  "Addison",
                  ", ",
                  "TX",
                  " ",
                  "75254",
                ],
                "href": "https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Array [
                "5100 Belt Line Road, STE 502",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "Addison",
                ", ",
                "TX",
                " ",
                "75254",
              ],
              "type": "a",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="restaurant-detail__phone"
                >
                  <a
                    href="tel:9723872337"
                  >
                    (972) 387-2337
                  </a>
                </div>,
                <div
                  className="restaurant-detail__twitter"
                >
                  <a
                    href="https://twitter.com/hopdoddy"
                    target="_blank"
                  >
                    @
                    hopdoddy
                  </a>
                </div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="tel:9723872337"
                  >
                    (972) 387-2337
                  </a>,
                  "className": "restaurant-detail__phone",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "(972) 387-2337",
                    "href": "tel:9723872337",
                  },
                  "ref": null,
                  "rendered": "(972) 387-2337",
                  "type": "a",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    href="https://twitter.com/hopdoddy"
                    target="_blank"
                  >
                    @
                    hopdoddy
                  </a>,
                  "className": "restaurant-detail__twitter",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "@",
                      "hopdoddy",
                    ],
                    "href": "https://twitter.com/hopdoddy",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Array [
                    "@",
                    "hopdoddy",
                  ],
                  "type": "a",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="restaurant-detail__map"
          >
            <withScriptjs(withGoogleMap(Component))
              containerElement={
                <div
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                />
              }
              googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places"
              isMarkerShow={true}
              isOpenMarkerInfo={false}
              lat={32.950787}
              lng={-96.821118}
              loadingElement={
                <div
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                />
              }
              mapElement={
                <div
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                />
              }
              onMarkerToggle={[Function]}
              restaurant={
                Object {
                  "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
                  "category": "Burgers",
                  "contact": Object {
                    "formattedPhone": "(972) 387-2337",
                    "phone": "9723872337",
                    "twitter": "hopdoddy",
                  },
                  "location": Object {
                    "address": "5100 Belt Line Road, STE 502",
                    "cc": "US",
                    "city": "Addison",
                    "country": "United States",
                    "crossStreet": "Dallas North Tollway",
                    "formattedAddress": Array [
                      "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
                      "Addison, TX 75254",
                      "United States",
                    ],
                    "lat": 32.950787,
                    "lng": -96.821118,
                    "postalCode": "75254",
                    "state": "TX",
                  },
                  "name": "Hopdoddy Burger Bar",
                }
              }
              zoom={13}
            />
          </div>,
          <div
            className="restaurant-detail__header"
          >
            <h2>
              Hopdoddy Burger Bar
            </h2>
            <div
              className="restaurant-detail__category"
            >
              Burgers
            </div>
          </div>,
          <div
            className="restaurant-detail__info"
          >
            <div
              className="restaurant-detail__address"
            >
              <a
                href="https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254"
                target="_blank"
              >
                5100 Belt Line Road, STE 502
                <br />
                Addison
                , 
                TX
                 
                75254
              </a>
            </div>
            <div>
              <div
                className="restaurant-detail__phone"
              >
                <a
                  href="tel:9723872337"
                >
                  (972) 387-2337
                </a>
              </div>
              <div
                className="restaurant-detail__twitter"
              >
                <a
                  href="https://twitter.com/hopdoddy"
                  target="_blank"
                >
                  @
                  hopdoddy
                </a>
              </div>
            </div>
          </div>,
        ],
        "className": "restaurant-detail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <withScriptjs(withGoogleMap(Component))
              containerElement={
                <div
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                />
              }
              googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places"
              isMarkerShow={true}
              isOpenMarkerInfo={false}
              lat={32.950787}
              lng={-96.821118}
              loadingElement={
                <div
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                />
              }
              mapElement={
                <div
                  style={
                    Object {
                      "height": "100%",
                    }
                  }
                />
              }
              onMarkerToggle={[Function]}
              restaurant={
                Object {
                  "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
                  "category": "Burgers",
                  "contact": Object {
                    "formattedPhone": "(972) 387-2337",
                    "phone": "9723872337",
                    "twitter": "hopdoddy",
                  },
                  "location": Object {
                    "address": "5100 Belt Line Road, STE 502",
                    "cc": "US",
                    "city": "Addison",
                    "country": "United States",
                    "crossStreet": "Dallas North Tollway",
                    "formattedAddress": Array [
                      "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
                      "Addison, TX 75254",
                      "United States",
                    ],
                    "lat": 32.950787,
                    "lng": -96.821118,
                    "postalCode": "75254",
                    "state": "TX",
                  },
                  "name": "Hopdoddy Burger Bar",
                }
              }
              zoom={13}
            />,
            "className": "restaurant-detail__map",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "containerElement": <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />,
              "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places",
              "isMarkerShow": true,
              "isOpenMarkerInfo": false,
              "lat": 32.950787,
              "lng": -96.821118,
              "loadingElement": <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />,
              "mapElement": <div
                style={
                  Object {
                    "height": "100%",
                  }
                }
              />,
              "onMarkerToggle": [Function],
              "restaurant": Object {
                "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
                "category": "Burgers",
                "contact": Object {
                  "formattedPhone": "(972) 387-2337",
                  "phone": "9723872337",
                  "twitter": "hopdoddy",
                },
                "location": Object {
                  "address": "5100 Belt Line Road, STE 502",
                  "cc": "US",
                  "city": "Addison",
                  "country": "United States",
                  "crossStreet": "Dallas North Tollway",
                  "formattedAddress": Array [
                    "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
                    "Addison, TX 75254",
                    "United States",
                  ],
                  "lat": 32.950787,
                  "lng": -96.821118,
                  "postalCode": "75254",
                  "state": "TX",
                },
                "name": "Hopdoddy Burger Bar",
              },
              "zoom": 13,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Hopdoddy Burger Bar
              </h2>,
              <div
                className="restaurant-detail__category"
              >
                Burgers
              </div>,
            ],
            "className": "restaurant-detail__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hopdoddy Burger Bar",
              },
              "ref": null,
              "rendered": "Hopdoddy Burger Bar",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Burgers",
                "className": "restaurant-detail__category",
              },
              "ref": null,
              "rendered": "Burgers",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="restaurant-detail__address"
              >
                <a
                  href="https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254"
                  target="_blank"
                >
                  5100 Belt Line Road, STE 502
                  <br />
                  Addison
                  , 
                  TX
                   
                  75254
                </a>
              </div>,
              <div>
                <div
                  className="restaurant-detail__phone"
                >
                  <a
                    href="tel:9723872337"
                  >
                    (972) 387-2337
                  </a>
                </div>
                <div
                  className="restaurant-detail__twitter"
                >
                  <a
                    href="https://twitter.com/hopdoddy"
                    target="_blank"
                  >
                    @
                    hopdoddy
                  </a>
                </div>
              </div>,
            ],
            "className": "restaurant-detail__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254"
                  target="_blank"
                >
                  5100 Belt Line Road, STE 502
                  <br />
                  Addison
                  , 
                  TX
                   
                  75254
                </a>,
                "className": "restaurant-detail__address",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "5100 Belt Line Road, STE 502",
                    <br />,
                    "Addison",
                    ", ",
                    "TX",
                    " ",
                    "75254",
                  ],
                  "href": "https://www.google.com/maps/dir/?api=1&destination=Hopdoddy%20Burger%20Bar%205100%20Belt%20Line%20Road%2C%20STE%20502%20Addison%20TX%2075254",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Array [
                  "5100 Belt Line Road, STE 502",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "Addison",
                  ", ",
                  "TX",
                  " ",
                  "75254",
                ],
                "type": "a",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="restaurant-detail__phone"
                  >
                    <a
                      href="tel:9723872337"
                    >
                      (972) 387-2337
                    </a>
                  </div>,
                  <div
                    className="restaurant-detail__twitter"
                  >
                    <a
                      href="https://twitter.com/hopdoddy"
                      target="_blank"
                    >
                      @
                      hopdoddy
                    </a>
                  </div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="tel:9723872337"
                    >
                      (972) 387-2337
                    </a>,
                    "className": "restaurant-detail__phone",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "(972) 387-2337",
                      "href": "tel:9723872337",
                    },
                    "ref": null,
                    "rendered": "(972) 387-2337",
                    "type": "a",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      href="https://twitter.com/hopdoddy"
                      target="_blank"
                    >
                      @
                      hopdoddy
                    </a>,
                    "className": "restaurant-detail__twitter",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "@",
                        "hopdoddy",
                      ],
                      "href": "https://twitter.com/hopdoddy",
                      "target": "_blank",
                    },
                    "ref": null,
                    "rendered": Array [
                      "@",
                      "hopdoddy",
                    ],
                    "type": "a",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
